osbrain/tests/test_agent_async_requests.py

```diff
@@ -139,8 +139,8 @@ def test_unknown(nsproxy, server_client_late_reply_return):
 
     # Manually remove the pending request before it is received
     client.set_attr(_pending_requests={})
-    time.sleep(1.1)
-    assert server.get_attr('received') == ['foo']
+    server.unsafe.set_attr(blocked=False)
+    assert wait_agent_attr(server, value=['foo'], timeout=.1)
     assert client.get_attr('received') == []
```

Comment:
I would rather use `not wait_agent_attr(client, length=1, timeout=.1)`. You can also put the logger assertion before the client assertion.

Maybe you can move the `assert wait_agent_attr(server...)` up just after the `client.send()`.

Reply:
If you don't mind, I like the order as it is. The code looks better with `logger_received` at the end IMO.

+1 to the `not wait_agent_atttr(length=1)` thing, though. I think I'll use it in other places as well.